source/dagger/gameplay/team_game/character_controller_states/idle_state.cpp

```diff
@@ -0,0 +1,43 @@
+#include "idle_state.h"
+
+#include "core/engine.h"
+#include "core/input/inputs.h"
+
+#include "gameplay/team_game/character_controller.h"
+
+void IdleStateSystem::OnInitialize(Registry& registry_, Entity entity_)
+{
+	InputReceiver& receiver = registry_.get<InputReceiver>(entity_);
```

Comment:
I'd actually propose something a bit different, if there's interest. Instead of making engine-level systems for states, I'd go for creating a `FiniteStateMachine` class that will basically do the same thing that the engine does (so, have a list of systems and cycle through them, but with the exception of only cycling through the active ones). Then both of these systems could be translated to a `State` base class that does the same thing, but isn't on the same level as the actual systems. Instead, the base class (`FiniteStateMachine`) could be that and hold all of that state within.

What I'm afraid of is polluting the systems loop with basically idling states from different state machines, instead of having just a `PlayerBehaviourSystem` which, internally, handles a sub-engine, and then an `AIBehaviourSystem`, etc,

What do you think?